pkg/diagnostics/systemd/systemd.go

```diff
@@ -0,0 +1,287 @@
+package systemd
+
+import (
+	"regexp"
+
+	"fmt"
+	"github.com/openshift/origin/pkg/diagnostics/log"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+)
+
+type logEntry struct {
+	Message   string `json:"MESSAGE"`
+	TimeStamp string `json:"__REALTIME_TIMESTAMP"` // epoch + ns
+}
+
+// logMatcher provides a regex for scanning log messages and parameteres to interpret them when found.
+type logMatcher struct {
+	Regexp         *regexp.Regexp
+	Level          log.Level
+	Id             string
+	Interpretation string // log with above level+id if it's simple
+	KeepAfterMatch bool   // usually note only first matched entry, ignore rest
+	// Interpret, if provided, runs custom logic against a match. Its return bool is the equivalent of KeepAfterMatch.
+	Interpret func(
+		entry *logEntry,
+		matches []string,
+		r types.DiagnosticResult,
+	) bool
+}
+
+type unitSpec struct {
+	Name        string
+	StartMatch  *regexp.Regexp // regex to look for in log messages indicating startup
+	LogMatchers []logMatcher   // suspect log patterns to check for - checked in order
+}
+
+//
+// -------- These are things that feed into the diagnostics definitions -----------
+//
+
+// Reusable log matchers:
+var badImageTemplate = logMatcher{
+	Regexp: regexp.MustCompile(`Unable to find an image for .* due to an error processing the format: %!v\\(MISSING\\)`),
+	Level:  log.InfoLevel,
+	Interpretation: `
+This error indicates the openshift command was given the --images flag
+with an invalid format variable. Valid formats can include (literally)
+${component} and ${version}; any others will cause this error.
+This could be a typo or you might be intending to hardcode something,
+such as a version which should be specified as e.g. v3.0, not ${v3.0}.
+Note that the --images flag may be supplied via the master, node, or
+"openshift admin registry/router" invocations and should usually
+be the same for each.`,
+}
+
+// captures for logMatcher Interpret functions to store state between matches
+var tlsClientErrorSeen map[string]bool
+
+// TODO, these should probably be split into interfaces similar to Diagnostic
+// Specify what units we can check and what to look for and say about it
+var unitLogSpecs = []*unitSpec{
+	{
+		Name:       "openshift-master",
+		StartMatch: regexp.MustCompile("Starting master on"),
+		LogMatchers: []logMatcher{
+			badImageTemplate,
+			{
+				Regexp:         regexp.MustCompile("Unable to decode an event from the watch stream: local error: unexpected message"),
+				Level:          log.InfoLevel,
+				Id:             "sdLogOMIgnore",
+				Interpretation: "You can safely ignore this message.",
+			},
+			{
+				Regexp: regexp.MustCompile("HTTP probe error: Get .*/healthz: dial tcp .*:10250: connection refused"),
```

Comment:
You're right, I'll look into whether it would help to check for something similar on the node side instead. Although node status checks already at least let you know something is wrong.